src/android/UploadTask.java

```diff
@@ -160,23 +176,68 @@ private static ForegroundInfo getForegroundInfo(final Context context) {
                     .setTicker(notificationTitle)
                     .setSmallIcon(notificationIconRes)
                     .setColor(Color.rgb(57, 100, 150))
+                    // Notify device that the notification is for an ongoing process
                     .setOngoing(true)
                     .setProgress(100, (int) (totalProgress * 100f), false)
                     .build();
 
+            // Not allow user to clear
+            notification.flags |= Notification.FLAG_NO_CLEAR;
+            // Notify device that the notification is for an ongoing process
+            // Some device support this flag
+            notification.flags |= Notification.FLAG_ONGOING_EVENT;
+            // Notify device that the notification is for a currently running foreground service
+            notification.flags |= Notification.FLAG_FOREGROUND_SERVICE;
+
             cachedInfo = new ForegroundInfo(notificationId, notification);
             return cachedInfo;
         }
+
+        // Foreground notification used to tell user that there is some images left to be uploaded
+        public static ForegroundInfo getRetryForegroundInfo(final Context context) {
+            Notification retryNotification = new NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID)
+                    .setContentTitle(notificationRetryTitle)
+                    .setTicker(notificationRetryTitle)
+                    .setContentText(notificationRetryText)
+                    .setSmallIcon(notificationIconRes)
+                    .setColor(Color.rgb(57, 100, 150))
+                    .build();
+
+            retryCachedInfo = new ForegroundInfo(notificationId, retryNotification);
+            return retryCachedInfo;
+        }
+
+        @Nullable
+        @Override
+        public IBinder onBind(Intent intent) {
+            return null;
+        }
     }
     // </editor-fold>
 
     private static OkHttpClient httpClient;
 
     private Call currentCall;
 
+    private static int concurency = 1;
+
+    private static Semaphore concurrentUploads = new Semaphore(concurency, true);
+
+    private static Semaphore concurrencyLock = new Semaphore(1);
```

Comment:
```suggestion
    private static int concurency = 1;
    private static Semaphore concurrentUploads = new Semaphore(concurency, true);
    private static Semaphore concurrencyLock = new Semaphore(1);
```
we can group related attributes, for example here, all 3 are related to concurrency